lib/ansible/plugins/lookup/azure_keyvault_secret.py

```diff
@@ -0,0 +1,123 @@
+"""
+
+EXAMPLE = """
+- debug: msg="the value of this secret is {{lookup('azure_keyvault_secret','testSecret/version')}}"
+
+- debug: msg="the value of this secret is {{lookup('azure_keyvault_secret','testSecret/version',vault_url='https://myvault.vault.azure.net', cliend_id='123456789', key='abcdefg', tenant_id='uvwxyz')}}"
+"""
+
+RETURN = """
+  _raw:
+    description: secret content
+"""
+
+from ansible.errors import AnsibleError, AnsibleParserError
+from ansible.plugins.lookup import LookupBase
+import requests
+
+TOKEN_ACQUIRED = False
+
+token_params = {
+    'api-version': '2018-02-01',
+    'resource': 'https://vault.azure.net'
+}
+token_headers = {
+    'Metadata': 'true'
+}
+token = None
+try:
+    token_res = requests.get('http://169.254.169.254/metadata/identity/oauth2/token', params=token_params, headers=token_headers)
+    token = token_res.json()["access_token"]
+    TOKEN_ACQUIRED = True
+except requests.exceptions.RequestException as e:
+    print('Unable to fetch MSI token. Will use service principal if provided.')
+    TOKEN_ACQUIRED = False
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables, **kwargs):
+
+        ret = []
+        vault_url = kwargs.pop('vault_url', None)
+        if TOKEN_ACQUIRED:
+            secret_params = {'api-version': '2016-10-01'}
+            secret_headers = {'Authorization': 'Bearer ' + token}
+            for term in terms:
+                try:
+                    secret_res = requests.get(vault_url + 'secrets/' + term, params=secret_params, headers=secret_headers)
+                    ret.append(secret_res.json()["value"])
+                except requests.exceptions.RequestException as e:
+                    raise AnsibleError('Failed to fetch secret: ' + term + ' via MSI endpoint.')
+                except KeyError:
+                    raise AnsibleError('Failed to fetch secret ' + term + '.')
+            return ret
+        else:
+            import logging
+            logging.getLogger('msrestazure.azure_active_directory').addHandler(logging.NullHandler())
+            logging.getLogger('msrest.service_client').addHandler(logging.NullHandler())
+
+            try:
+                from azure.common.credentials import ServicePrincipalCredentials
+                from azure.keyvault import KeyVaultClient
+                from msrest.exceptions import AuthenticationError, ClientRequestError
+                from azure.keyvault.models.key_vault_error import KeyVaultErrorException
+            except ImportError:
+                raise AnsibleError('The azure_keyvault_secret lookup plugin requires azure.keyvault and azure.common.credentials to be installed.')
+
+            client_id = kwargs.pop('client_id', None)
+            key = kwargs.pop('key', None)
+            tenant_id = kwargs.pop('tenant_id', None)
+
+            try:
+                credentials = ServicePrincipalCredentials(
```

Comment:
this should not work. please refer to code https://github.com/VSChina/ansible/blob/3defde44cd7f49d624ed7d74d4dbec70c8273297/lib/ansible/modules/cloud/azure/azure_rm_keyvaultsecret.py#L136-L153